packages/amplify-graphql-api-construct/src/internal/construct-exports.ts

```diff
@@ -17,6 +17,14 @@ import { AmplifyGraphqlApiResources, FunctionSlot } from '../types';
 import { AmplifyDynamoDbTableWrapper } from '../amplify-dynamodb-table-wrapper';
 import { walkAndProcessNodes } from './construct-tree';
 
+/**
+ * Check if a resource is implementing table interface
+ * @param table table resource
+ * @returns whether the resource is a ITable or not
+ */
+function isITable(table: any): table is ITable {
```

Comment:
Does the table in `tableFromAttr` below actually conform to ITable, or does it only contain the `tableArn`? If the former, this type predicate is inaccurate, since it is both missing required fields and requiring `tableArn` which is an optional field. ([Ref](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_dynamodb.ITable.html)) If the latter, we shouldn't be putting it in a structure which is declaring `ITable` since it doesn't actually conform to that interface.

Reply:
The `tableFromAttr` is defined as `ITable` [here](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-dynamodb-table-construct/index.ts#L37). It is created via the CDK Table construct call `Table.fromTableAttributes` ([source code](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-dynamodb-table-construct/index.ts#L106-L109)).

Reply:
I've added all required field of `ITable` in the check